tests/system/src/test/java/test/javafx/scene/web/WebPageTest.java

```diff
@@ -0,0 +1,148 @@
+
+import com.sun.webkit.WebPage;
+import com.sun.webkit.WebPageShim;
+
+import static javafx.concurrent.Worker.State.SUCCEEDED;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class WebPageTest {
+    private static final CountDownLatch launchLatch = new CountDownLatch(1);
+
+    // Maintain one application instance
+    static WebPageTestApp webPageTestApp;
+
+    private WebView webView;
+
+    public static class WebPageTestApp extends Application {
+        Stage primaryStage = null;
+
+        @Override
+        public void init() {
+            WebPageTest.webPageTestApp = this;
+        }
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            Platform.setImplicitExit(false);
+            this.primaryStage = primaryStage;
+            launchLatch.countDown();
+        }
+    }
+
+    @BeforeClass
+    public static void setupOnce() {
+        // Start the Test Application
+        new Thread(() -> Application.launch(WebPageTestApp.class, (String[])null)).start();
+
+        assertTrue("Timeout waiting for FX runtime to start", Util.await(launchLatch));
+    }
+
+    @AfterClass
+    public static void tearDownOnce() {
+        Platform.exit();
+    }
+
+    @Before
+    public void setupTestObjects() {
+        Platform.runLater(() -> {
+            webView = new WebView();
+            webPageTestApp.primaryStage.setScene(new Scene(webView));
+            webPageTestApp.primaryStage.show();
+        });
+    }
+
+    /**
+     * @test
+     * @bug 8260257
+     * summary Checks if scrolling is possible
+     */
+    @Test public void testScroll() {
+        final CountDownLatch webViewStateLatch = new CountDownLatch(1);
+        final String htmlContent = "\n"
+            + "<html>\n"
+            + "<body style='height:1500px'>\n"
+            + "<p id='test'>Fail</p>\n"
+            + "<script>\n"
+            + "window.onscroll = function() {scrollFunc()};\n"
+            + "function scrollFunc() {\n"
+            + "document.getElementById('test').innerHTML = 'Pass';\n"
+            + "}\n"
+            + "</script>\n"
+            + "</body>\n"
+            + "</html>";
+
+        Util.runAndWait(() -> {
+            assertNotNull(webView);
+            webView.getEngine().getLoadWorker().stateProperty().
+                addListener((observable, oldValue, newValue) -> {
+                if (newValue == SUCCEEDED) {
+                    webView.requestFocus();
+                }
+            });
+
+            webView.focusedProperty().
+                addListener((observable, oldValue, newValue) -> {
+                if (newValue) {
+                    webViewStateLatch.countDown();
+                }
+            });
+
+            webView.getEngine().loadContent(htmlContent);
+        });
+
+        assertTrue("Timeout when waiting for focus change ", Util.await(webViewStateLatch));
+
```

Comment:
I needed to add a `sleep(500)` here.

Reply:
I've used `Util.sleep(1000)` instead to make it similar to the other system web tests.